### ISSUE TITLE
IMU Coordinate system is wrong in SDK Document.

### ISSUE THREAD

**Issue by saikishor** [slightech/MYNT-EYE-D-SDK#58]
I am not sure, which is right, but the coordinate system mentioned in the document is completely different from the one in the ros wrapper. Please find the below for details:

![The image from ROS Wrapper](https://user-images.githubusercontent.com/10082428/68681116-19bded00-0563-11ea-9833-0a56650465d9.png)

![Coordinate system from the Docs](https://user-images.githubusercontent.com/10082428/68681162-35c18e80-0563-11ea-96f7-a00e8306631d.png)
https://buildmedia.readthedocs.org/media/pdf/mynt-eye-d-sdk/latest/mynt-eye-d-sdk.pdf

I believe the ROS one is appropriate as the one in the docs seems to have the left handed one, but is mentioned as right handed. 

Can someone please confirm which is right?

@yangruixuan @joinAero @harjeb any idea on this?

**Comment by KalmanXiong** [comment]
IMU coordinate system in document is right. This coordinate system based on the datasheet of the sensor(BMI088). However, the Ros one based on the map. I will report the issue to our team  and decide that whether we should unified the two coordinate systems.

**Comment by saikishor** [comment]
@KalmanXiong 
I am not sure if it if right, because it doesn't represent the right hand system. As per your advice, I have checked the datasheet of BMI088 (https://ae-bst.resource.bosch.com/media/_tech/media/datasheets/BST-BMI088-DS001.pdf). I found the below image, in the page 44 of the document.
![image](https://user-images.githubusercontent.com/10082428/68743618-2853e500-05f3-11ea-872d-013f089e9487.png)

As per which aligning both and Z and X coordinate system the Y-axis should point **downwards** instead of **upwards** as per your documentation diagram. Then it makes sense and you would be able to obtain the same transformation in the ROS too.

Please kindly share your opinion on this. Thank you.

Moreover, I would kindly request you to report your team to fix the transformation of all the frames in the ROS, because all the frames are added to origin simply with respective rotations, but not a proper translation.

**Comment by KalmanXiong** [comment]
In our document, we indeed show the IMU coordinate system as right-hand system as well

**Comment by saikishor** [comment]
@KalmanXiong I completely agree, I was just referring to the coordinate system in the image. Where the Y-axis in theory should point in opposite direction. Could you please review and confirm this?.

**Comment by KalmanXiong** [comment]
Please image that the coordinate system in the the datasheet of BMI088, flip up and down 180 degrees, and then turn 90 degrees counter clockwise. So you get the coordinate system in our doucment.

**Comment by KalmanXiong** [comment]
When you flip up and down the coordinate  system, keep the x-axis direction the same, and then  turn 90 degrees counter clockwise

**Comment by saikishor** [comment]
@KalmanXiong May be I am a bit confused, what do you mean exactly by flip up and down the coordinate system?. Rotate 180 degrees clockwise around Y-axis? or around X-axis?.  Moreover, in one way or another, I see one of the frame pointing opposite. Can you please confirm about the flip?.

**Comment by KalmanXiong** [comment]
> @KalmanXiong May be I am a bit confused, what do you mean exactly by flip up and down the coordinate system?. Rotate 180 degrees clockwise around Y-axis? or around X-axis?. Moreover, in one way or another, I see one of the frame pointing opposite. Can you please confirm about the flip?.

Rotate  the coordinate system 180 degrees clockwise around  X-axis, and then turn it  90 degrees counter clockwise around Z-axis.

**Comment by saikishor** [comment]
![image](https://user-images.githubusercontent.com/10082428/68754881-e2ede280-0607-11ea-9191-7413e326a064.png)

Ok got it, I made a simple sketch with the information. Now it makes sense. So as per the document image, the Y axis is will be pointing behind the camera right?. As shown in the above image. 

I was confused with the coordinate system in the documentation, I thought the Y is pointing towards the front side of the camera.

@KalmanXiong  Can you confirm this please?

**Comment by KalmanXiong** [comment]
![1573711907(1)](https://user-images.githubusercontent.com/37530151/68831240-bfe72f80-06e8-11ea-9741-dd90a6e2c0b8.png)
The image might could help you. I can confirm that the coordinate system in the documentation is correct.

**Comment by saikishor** [comment]
I am sorry but I couldn't agree with you, if you say this is right, then the documentation of IMU in MYNT-EYE-S is completely wrong, which I don't think so, I believe the coordinate system in the MYNT-EYE S IMU is Perfect. If you refer to the three IMU coordinate system, here (https://mynt-eye-s-sdk.readthedocs.io/en/latest/src/product/imu_coord_s2110.html, https://mynt-eye-s-sdk.readthedocs.io/en/latest/src/product/imu_coord_s2100.html, https://mynt-eye-s-sdk.readthedocs.io/en/latest/src/product/imu_coord_s1030.html). They all make sense and are right handed system.

But here, I couldn't think of a transformation for the frames, because it is represented by left-handed axis. I believe it should be something like:
![image](https://user-images.githubusercontent.com/10082428/68838809-a54c9080-06bf-11ea-8ad3-78674af44588.png)

If you believe that what you represent is true, please help me with a proper transformation to obtain that coordinate system. 

@KalmanXiong Please let me know something.

**Comment by harjeb** [comment]
Hi, have a look at this pic，the z axis is parallel with optic axis
![image2019-8-22_11-5-7](https://user-images.githubusercontent.com/18549414/68911439-faf06e00-078f-11ea-8118-2ee8efc0c938.png)

**Comment by saikishor** [comment]
@harjeb awesome where did you find the picture?.  yeah you have the imu orientation in the image, from this we can get the proper orientation of the frame?.
This means that X-axis coincides and Y points to the top of the camera and the Z-axis is parallel to the optic axis, but in opposite direction to global. This completely makes sense.

Could you please point me to the source?

**Comment by KalmanXiong** [comment]
What we discussed above is MYNT-EYE-D. why did you talk about  MYNT-EYE-S suddenly?

**Comment by saikishor** [comment]
@KalmanXiong Yeah, they don't have any relation, but what i was mentioning is all of those IMU's are right handed system and their coordinate system make sense, when talking about Mynt Eye D, it is a right hand system, but the coordinate system representation is not exactly Right handed, that's what i am pointing out above. 

I guess what @harjeb pointed out could be right. Do you know it's source?

**Comment by harjeb** [comment]
Hi @saikishor , this picture is internal data,  and I have confirmed it is correctly

**Comment by saikishor** [comment]
@harjeb Thanks for the update, would you be kind enough to share the similar picture with measurements, this will help me to define my frames in ROS, which is very important from sensor point of view, having an offset in frames creates a bigger problems in future. I hope you don't mind.

Once I am done with proper frames, I can create a PR to the repo.

**Comment by saikishor** [comment]
@harjeb From here, I know the placement in 2D, but It would be helpful if you can share the picture with the measurements from top view, so I can create perfect frames.

**Comment by saikishor** [comment]
> Hi, have a look at this pic，the z axis is parallel with optic axis
> ![image2019-8-22_11-5-7](https://user-images.githubusercontent.com/18549414/68911439-faf06e00-078f-11ea-8118-2ee8efc0c938.png)

Here, Where will exactly be the depth frame?

**Comment by saikishor** [comment]
@harjeb can you please comment on my request earlier?

**Comment by harjeb** [comment]
@saikishor Hi , there‘s no top view  right now

**Comment by saikishor** [comment]
@harjeb thanks for update. Do you know how deep is the sensor placed from the front end of the device.?

**Comment by saikishor** [comment]
@harjeb and @KalmanXiong We are planning to use Mynt Eye D cameras for our robots. Do you have an estimate, when these documents are available to public to have a proper frames in ROS.
Out of the topic, when can we also estimate support for multiple cameras?

**Comment by saikishor** [comment]
@harjeb  and @KalmanXiong Can you please comment on this?

**Comment by harjeb** [comment]
We still have problems to solve, it‘s hard to say when it will be ready.

**Comment by saikishor** [comment]
@harjeb Thanks for the update. :)

**Comment by harjeb** [comment]
We still have problems to solve, it‘s hard to say when it is ready.

**Comment by ahmetsaglam** [comment]
> @harjeb and @KalmanXiong We are planning to use Mynt Eye D cameras for our robots. Do you have an estimate, when these documents are available to public to have a proper frames in ROS.
> Out of the topic, when can we also estimate support for multiple cameras?

Hi all,
Is there any progress on the documentation for this specific matter? We are also planning to use Mynt Eye D cameras and IMU for our robots and I need this information immediately.

**Comment by RaviBeagle** [comment]
> Hi, have a look at this pic，the z axis is parallel with optic axis ![image2019-8-22_11-5-7](https://user-images.githubusercontent.com/18549414/68911439-faf06e00-078f-11ea-8118-2ee8efc0c938.png)

Is this final correct co-ordinate system of IMU ? Also z is parallel to optic axis and directed backward ?
